### PR TITLE
fix: empty selection in jsfile command

### PR DESCRIPTION
**Pull request by Pomroka**
Looks like awaiting for reading file is clearing selection. Took me some time to find why in `jscommand` i get selection and in `jsfile` selection is empty.
Plus fix on missing argument in Promise resolve warning

### CODE REVIEW

**Comment by Pomroka** [comment]
Added fix for #104 and #129  `selection` not working in next opened files after first.

File: main.ts [hunk 1]
```diff
@@ -101,6 +104,14 @@ export default class VimrcPlugin extends Plugin {
 		})
 	}
 
+	registerFileOpenEvent(){
+		// Record the position of selections in each file opened
+		let cmEditor = this.getCodeMirror(this.getActiveView());
+		CodeMirror.on(cmEditor, "cursorActivity", async (cm: any) => {
```
Comment (esm7) [inline main.ts:110]: Are you sure this happens just once on each CM instance? I'm not sure at all that Obsidian guarantees that this doesn't happen an infinite number of times on the same editor instance.
Maybe `on('codemirror', ...)` is a safer place to put this?

Comment (Pomroka) [reply main.ts]: with `on('codemirror', ...)` couldn't get it to work

**Comment by Laharah** [comment]
I can confirm that this PR fixes #129

**Comment by ShaySS** [comment]
Any chance this could get merged? This issue also affects the `:pasteinto` functionality and I desperately want a fix for that

**Comment by esm7** [comment]
Yes, I know it takes more time than it should, but this week!

**Pull request merged**